client/src/components/Search/TagsSearch/TagsSearch.js

```diff
@@ -73,19 +102,22 @@ class TagsFilter extends React.Component {
                 </div>
                 <div className="search-results">
                     <Tags selectable onSelect={this.handleSelectTag.bind(this)} tags={visibleSearchResults}/>
+                    {createNewTagDisplay}
                 </div>
             </div>
         );
     }
 
     async handleSelectTag(value, tag) {
         let { selectedTags } = this.state;
+        const { createdTags } = this.state;
+        const { variant, onChange } = this.props;
```

Comment:
This component also takes a few more props now so selecting/deselecting a Tag/Material can be managed by the parent (See `Filters` for an example).